pkg/alertmanager/statefulset.go

```diff
@@ -84,6 +85,7 @@ func makeStatefulSet(am *monitoringv1.Alertmanager, old *appsv1.StatefulSet, con
 	// do not transfer kubectl annotations to the statefulset so it is not
 	// pruned by kubectl
 	annotations := make(map[string]string)
+	annotations[sSetInputHashName] = inputHash
```

Comment:
it would be safer to move it after the for loop (in case someone puts the same annotation on the Alertmanager resource).

Reply:
I moved it.

Reply:
Nevertheless, unit tests are broken. I don't have time or skills to fix them.